plugins/inputs/prometheus/prometheus.go

```diff
@@ -63,6 +79,28 @@ func (p *Prometheus) Description() string {
 
 var ErrProtocolError = errors.New("prometheus protocol error")
 
+func (p *Prometheus) SplitHostAndPort(host string) (string, string) {
+	hostPort := ""
+	hostSplit := strings.Split(host, ":")
+	if len(hostSplit) == 2 {
+		host = hostSplit[0]
+		hostPort = hostSplit[1]
+	}
+	return host, hostPort
+}
+
+func (p *Prometheus) AddressToURL(u *url.URL, address string) string {
```

Comment:
This can be done with URL and its String() function.

Reply:
Done!